ev3dev/control/GyroBalancer.py

```diff
@@ -29,44 +29,48 @@
 
 import logging
 import time
+import json
+import queue
+import threading
 from collections import deque
 from ev3dev.power import PowerSupply
 from ev3dev.motor import LargeMotor, OUTPUT_A, OUTPUT_D
-from ev3dev.sensor.lego import GyroSensor, InfraredSensor, TouchSensor
+from ev3dev.sensor.lego import GyroSensor, TouchSensor
+from ev3dev.sound import Sound
+from collections import OrderedDict
 
 log = logging.getLogger(__name__)
 
-########################################################################
-# File I/O functions
-########################################################################
+# Constants
+RAD_PER_DEG = 3.14159265/180
 
+# EV3 Platform specific constants
 
-def FastRead(infile):
-    """Function for fast reading from sensor files."""
-    infile.seek(0)
-    return(int(infile.read().decode().strip()))
+# For the LEGO EV3 Gyro in Rate mode, 1 unit = 1 deg/s
+DEG_PER_SEC_PER_RAW_GYRO_UNIT = 1
 
+# Express the above as the rate in rad/s per gyro unit
+RAD_PER_SEC_PER_RAW_GYRO_UNIT = DEG_PER_SEC_PER_RAW_GYRO_UNIT * RAD_PER_DEG
 
-def FastWrite(outfile, value):
-    """Function for fast writing to motor files."""
-    outfile.truncate(0)
-    outfile.write(str(int(value)))
-    outfile.flush()
+# For the LEGO EV3 Large Motor 1 unit = 1 deg
+DEG_PER_RAW_MOTOR_UNIT = 1
 
+# Express the above as the angle in rad per motor unit
+RAD_PER_RAW_MOTOR_UNIT = DEG_PER_RAW_MOTOR_UNIT * RAD_PER_DEG
 
-def SetDuty(motorDutyFileHandle, duty, frictionOffset, voltageCompensation):
-    """Function to set the duty cycle of the motors."""
-    # Compensate for nominal voltage and round the input
-    dutyInt = int(round(duty*voltageCompensation))
+# On the EV3, "1% speed" corresponds to 1.7 RPM (if speed
+# control were enabled).
+RPM_PER_PERCENT_SPEED = 1.7
```

Comment:
Doesn’t this assume LargeMotor is in use? Maybe make this dynamic based on the motor class that is being used.

Reply:
Yes. the factor is 2.67 for medium motor.